### PR TITLE
chore(ui): Migrate the dialog component to UI package

### LINKED ISSUE #28953: Separate the Dailog component

### PR DESCRIPTION
**Pull request by nicobytes**
### Parent Issue

https://github.com/dotCMS/core/issues/27458

### Proposed Objective

Migrate the Image AI dialog component from the `@dotcms/block-editor` to the `@dotcms/ui`  package so that you can reuse this component's dialog in other components, such as Binary Field.

Screenshot 2024-06-17 at 11 19 51 AM

### Acceptance Criteria

- The dialog within the block editor should work fine after this migration.

This PR fixes: #28953

### CODE REVIEW

File: core-web/libs/edit-content/src/lib/fields/dot-edit-content-category-field/dot-edit-content-category-field.component.html [hunk 1]
```diff
@@ -1,14 +1,14 @@
...
-    @for (category of values; track category.id) {
-    <p-chip
-        [pTooltip]="category.value"
-        [removable]="true"
-        [label]="category.value"
-        tooltipPosition="top"
-        styleClass="p-chip-sm" />
-    }
-</div>
+    <div class="dot-category-field__categories" data-testId="category-chip-list">
```
Comment (oidacra) [inline core-web/libs/edit-content/src/lib/fields/dot-edit-content-category-field/dot-edit-content-category-field.component.html:2]: I think this file does not belong to the PR.

File: core-web/libs/ui/src/lib/components/dot-ai-image-prompt/components/ai-image-prompt-form/ai-image-prompt-form.component.spec.ts [hunk 2]
```diff
@@ -121,28 +121,4 @@ describe('AiImagePromptFormComponent', () => {
 
         expect(spectator.query(byTestId('prompt-label')).classList).not.toContain(REQUIRED_CLASS);
     });
-
```
Comment (hmoreras) [inline core-web/libs/ui/src/lib/components/dot-ai-image-prompt/components/ai-image-prompt-form/ai-image-prompt-form.component.spec.ts:124]: why removed ?

**Review by hmoreras — CHANGES_REQUESTED** [review]

**Review by fmontes — APPROVED** [review]

**Review by hmoreras — APPROVED** [review]

**Review by oidacra — APPROVED** [review]
